packages/opentelemetry-web/src/utils.ts

```diff
@@ -0,0 +1,176 @@
+    // for example "secureConnectionStart" is 0 which makes the events to be wrongly interpreted
+    if (entries[performanceName] === 0) {
+      return undefined;
+    }
+    span.addEvent(performanceName, entries[performanceName]);
+    return span;
+  }
+  return undefined;
+}
+
+/**
+ * Get closes performance resource ignoring the resources that has been already used.
+ * @param span
+ * @param resources
+ * @param ignoredResources
+ */
+export function getResource(
+  span: tracing.Span,
+  resources: PerformanceResourceTiming[],
+  ignoredResources?: PerformanceResourceTiming[]
+): PerformanceResourceTiming | undefined {
+  const filteredResources = filterResourcesForSpan(
+    span,
+    resources,
+    ignoredResources
+  );
+
+  if (filteredResources.length === 0) {
+    return undefined;
+  }
+  if (filteredResources.length === 1) {
+    return filteredResources[0];
+  }
+  const sorted = filteredResources.slice().sort((a, b) => {
+    const valueA = a[PTN.FETCH_START];
+    const valueB = b[PTN.FETCH_START];
+    if (valueA > valueB) {
+      return 1;
+    } else if (valueA < valueB) {
+      return -1;
+    }
+    return 0;
+  });
+  return sorted[0];
+}
+
+/**
+ * Filter all resources that has started and finished according to span start time and end time.
+ *     It will return the closest resource to a start time
+ * @param span
+ * @param resources
+ * @param ignoredResources
+ */
+function filterResourcesForSpan(
+  span: tracing.Span,
+  resources: PerformanceResourceTiming[],
+  ignoredResources?: PerformanceResourceTiming[]
+) {
+  const startTime = hrTimeToNanoseconds(span.startTime);
+  const nowTime = hrTimeToNanoseconds(hrTime());
+
+  const spanUrl = span.name;
+  let filteredResource = resources.filter(resource => {
+    const resourceStartTime = hrTimeToNanoseconds(
+      timeInputToHrTime(resource[PTN.FETCH_START])
+    );
+    const resourceEndTime = hrTimeToNanoseconds(
+      timeInputToHrTime(resource[PTN.RESPONSE_END])
+    );
+
+    return (
+      resource.initiatorType.toLowerCase() === 'xmlhttprequest' &&
+      resource.name === spanUrl &&
+      resourceStartTime >= startTime &&
+      resourceEndTime <= nowTime
+    );
+  });
+
+  if (filteredResource.length > 0) {
+    filteredResource = filteredResource.filter(resource => {
+      if (!ignoredResources) {
+        return true;
+      }
+      for (let i = ignoredResources.length - 1; i >= 0; i--) {
+        if (ignoredResources[i] === resource) {
+          return false;
+        }
+      }
+      return true;
+    });
+  }
+
+  return filteredResource;
+}
+
+/**
+ * Parses url using anchor element
+ * @param url
+ */
+export function parseUrl(url: string): HTMLAnchorElement {
+  const a = document.createElement('a');
+  a.href = url;
+  return a;
+}
+
+/**
+ * Check if both url might have the same origin or
+ *     if {@param urlToMatch} is part of {@param url}
+ * @param url
+ * @param urlToMatch
+ */
+export function sameOriginOrUrlMatches(
```

Comment:
I actually think that for the list of URLs, we just check their regex only.

The "same origin" notion to me represents only whether the request is the same origin as the server that sent down the HTML, i.e. in the context of the "same origin policy" browser security model where same origin requests don't need CORS preflights.